src/parser.rs

```diff
@@ -198,10 +199,94 @@ fn parse_unknown(input: (&[u8], usize)) -> IResult<(&[u8], usize), MessageKind>
     Ok((input, MessageKind::Unknown))
 }
 
+pub fn decode_id_13_field(f: u16) -> u16 {
+    let mut hex_gillham = 0;
+    if f & 0x1000 != 0 {
+        hex_gillham |= 0x0010;
+    } // Bit 12 = C1
+    if f & 0x0800 != 0 {
+        hex_gillham |= 0x1000;
+    } // Bit 11 = A1
+    if f & 0x0400 != 0 {
+        hex_gillham |= 0x0020;
+    } // Bit 10 = C2
+    if f & 0x0200 != 0 {
+        hex_gillham |= 0x2000;
+    } // Bit  9 = A2
+    if f & 0x0100 != 0 {
+        hex_gillham |= 0x0040;
+    } // Bit  8 = C4
+    if f & 0x0080 != 0 {
+        hex_gillham |= 0x4000;
+    } // Bit  7 = A4
+      //if (ID13Field & 0x0040) {hexGillham |= 0x0800;} // Bit  6 = X  or M
```

Comment:
Any reason for not handling this case?

Reply:
That bit appears to be a "don't care" bit.  From https://mode-s.org/decode/content/mode-s/3-surveillance.html:
 
> The 13-bit identity code encodes the 4 octal digit squawk code (from 0000 to 7777). The structure of this field is shown as follows:
> 
> ```
> +----+----+----+----+----+----+---+----+----+----+----+----+----+
> | C1 | A1 | C2 | A2 | C4 | A4 | X | B1 | D1 | B2 | D2 | B4 | D4 |
> +----+----+----+----+----+----+---+----+----+----+----+----+----+
> ```
> 
> The binary representation of the octal digit is:
> ```
> A4 A2 A1 | B4 B2 B1 | C4 C2 C1 | D4 D2 D1
> ```
>